src/ray/gcs/gcs_server/gcs_actor_manager.cc

```diff
@@ -0,0 +1,299 @@
+
+WorkerID GcsActor::GetWorkerID() const {
+  const auto &address = actor_table_data_.address();
+  if (address.worker_id().empty()) {
+    return WorkerID::Nil();
+  }
+  return WorkerID::FromBinary(address.worker_id());
+}
+
+void GcsActor::UpdateState(rpc::ActorTableData::ActorState state) {
+  actor_table_data_.set_state(state);
+}
+
+rpc::ActorTableData::ActorState GcsActor::GetState() const {
+  return actor_table_data_.state();
+}
+
+ActorID GcsActor::GetActorID() const {
+  return ActorID::FromBinary(actor_table_data_.actor_id());
+}
+
+TaskSpecification GcsActor::GetCreationTaskSpecification() const {
+  const auto &task_spec = actor_table_data_.task_spec();
+  return TaskSpecification(task_spec);
+}
+
+const rpc::ActorTableData &GcsActor::GetActorTableData() const {
+  return actor_table_data_;
+}
+
+rpc::ActorTableData *GcsActor::GetMutableActorTableData() { return &actor_table_data_; }
+
+/////////////////////////////////////////////////////////////////////////////////////////
+GcsActorManager::GcsActorManager(boost::asio::io_context &io_context,
+                                 gcs::ActorInfoAccessor &actor_info_accessor,
+                                 gcs::GcsNodeManager &gcs_node_manager,
+                                 LeaseClientFactoryFn lease_client_factory,
+                                 rpc::ClientFactoryFn client_factory)
+    : actor_info_accessor_(actor_info_accessor),
+      gcs_actor_scheduler_(new gcs::GcsActorScheduler(
+          io_context, actor_info_accessor, gcs_node_manager,
+          [this](std::shared_ptr<GcsActor> actor) {
+            // When there are no available nodes to schedule the actor the
+            // gcs_actor_scheduler will treat it as failed and invoke this handler. In
+            // this case, the actor should be appended to the `pending_actors_` and wait
+            // for the registration of new node.
+            pending_actors_.emplace(std::move(actor));
+          },
+          [this](std::shared_ptr<GcsActor> actor) {
+            auto worker_id = actor->GetWorkerID();
+            RAY_CHECK(!worker_id.IsNil());
+            RAY_CHECK(worker_to_created_actor_.emplace(worker_id, actor).second);
+
+            auto actor_id = actor->GetActorID();
+            auto node_id = actor->GetNodeID();
+            RAY_CHECK(!node_id.IsNil());
+            RAY_CHECK(node_to_created_actors_[node_id].emplace(actor_id, actor).second);
+
+            actor->UpdateState(rpc::ActorTableData::ALIVE);
+            auto actor_table_data =
+                std::make_shared<rpc::ActorTableData>(actor->GetActorTableData());
+            // The backend storage is reliable in the future, so the status must be ok.
+            RAY_CHECK_OK(
+                actor_info_accessor_.AsyncUpdate(actor_id, actor_table_data, nullptr));
+          },
+          std::move(lease_client_factory), std::move(client_factory))) {
```

Comment:
> Probably not necessary to do std::move since this is a one-time constructor.

I think it's necessary to move anonymous functions sometimes. In fact, we have done some internal tests and found that the copy of anonymous functions has a certain impact on performance, especially when there are some variables captured in lambda.